### ISSUE TITLE
Incorrect headers on keyrequest API endpoint

### ISSUE THREAD

**Issue by jvens** [FIRST-Tech-Challenge/scorekeeper#558]
**Describe the bug**
The CORS headers seem to be set incorrectly on the `keyrequest` API endpoint.  I am able to access all of the other API endpoints I have tested except for `keyrequest`.  I suspect it has something to do with the way `post` vs. `get` is handled in the server.

**To Reproduce**
Code to reproduce:
~~~javascript
axios.post('http://localhost/api/v1/keyrequest/?name=app', {})
    .then(response => console.log(response))
    .catch(err => console.error(err))
~~~

Error reported by chrome:
~~~javascript
Access to XMLHttpRequest at 'http://localhost/api/v1/keyrequest/?name=app' from origin 'http://localhost:8080' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: The 'Access-Control-Allow-Origin' header contains multiple values '*, *', but only one is allowed.
~~~

I believe the issue is that the `Access-Control-Allow-Origin` header is set twice causing the preflight to fail in chrome.

**Expected behavior**
Receive an API Key

**Screenshots**
N/A

**Device (please complete the following information):**
Tested on both
 - Chrome Version 97.0.4692.99
 - Electron 17.0.0
 - FTC Live Version 3.2.4

**Additional context**
Here is the header associated with the request.  As you can see several of the headers are duplicated in it.
~~~
HTTP/1.1 200 OK
Date: Wed, 02 Feb 2022 05:07:21 GMT
Content-Language: en
Set-Cookie: JSESSIONID=node01sf62ubehbrznjnpi6ztqrd1a108.node0; Path=/api
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Access-Control-Allow-Origin: *
Access-Control-Allow-Method: GET
Content-Language: en
Set-Cookie: FTCLIVE_JSESSIONID=node0wth3j4i4ghm1cv7c4mn66phe110.node0; Path=/
Access-Control-Allow-Origin: *
Access-Control-Allow-Method: GET
Allow: GET, HEAD, TRACE, OPTIONS
Content-Length: 0
Server: Jetty(9.4.31.v20200723)
~~~

Here are the headers from the `/api/v1/version` endpoint which works correctly
~~~
HTTP/1.1 200 OK
Date: Wed, 02 Feb 2022 05:07:21 GMT
Content-Language: en
Set-Cookie: JSESSIONID=node012rtlsad5g68s1g03y3xex15w0109.node0; Path=/api
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Content-Type: application/json
Access-Control-Allow-Origin: *
Access-Control-Allow-Method: GET
Transfer-Encoding: chunked
Server: Jetty(9.4.31.v20200723)
~~~

**Comment by tvbarnette999** [comment]
The CORS headers for OPTIONS pre-flights on all api routes should now be correctly set in v4.0.7 and not duplicated anymore.
Allow-Methods have been fixed to properly return "POST" for the keyrequest endpoint

**Comment by jvens** [comment]
I just did a quick test and have only dug a little bit into this, but it seems to still be failing for me.  It looks like the headers are correct for the cross origin access, but now it is failing for a different but related reason.  Also, now the get endpoints seem to be failing too.  I think it might be related to this: https://developer.chrome.com/blog/referrer-policy-new-chrome-default/.  I included all the information below, but if you would prefer I spin this off into a new ticket I am happy to do so.  I will likely have some time this weekend to possibly dig into this more and see if I can identify exactly what the cause is as well as a solution.

For my test setup I have the score system running at `http://localhost:8080` and I have a simple python webserver running at `http://localhost:8000`.  From the python server I am hosting the following file:
```
<html>
<head>
<script src="https://cdnjs.cloudflare.com/ajax/libs/axios/0.27.2/axios.min.js"></script>
<script>
axios.get('http://localhost:8080/api/v1/version', {})
    .then(response => console.log(response))
    .catch(err => console.error(err))

axios.post('http://localhost:8080/api/v1/keyrequest/?name=app', {})
    .then(response => console.log(response))
    .catch(err => console.error(err))
</script>
</head>
<body>
</body>
</html>
```

In the chrome console I get the following output when the page loads:
```
Access to XMLHttpRequest at 'http://localhost:8080/api/v1/version' from origin 'http://localhost:8000' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.
test.html:8 o {message: 'Network Error', name: 'AxiosError', code: 'ERR_NETWORK', config: {…}, request: XMLHttpRequest, …}code: "ERR_NETWORK"config: {transitional: {…}, transformRequest: Array(1), transformResponse: Array(1), timeout: 0, adapter: ƒ, …}message: "Network Error"name: "AxiosError"request: XMLHttpRequest {data: undefined, onreadystatechange: null, readyState: 4, timeout: 0, withCredentials: false, …}response: XMLHttpRequest {data: undefined, onreadystatechange: null, readyState: 4, timeout: 0, withCredentials: false, …}[[Prototype]]: Error
(anonymous) @ test.html:8
Promise.catch (async)
(anonymous) @ test.html:8
VM149:1          GET http://localhost:8080/api/v1/version net::ERR_FAILED 302
(anonymous) @ VM149:1
(anonymous) @ isAxiosError.js:12
e.exports @ isAxiosError.js:12
e.exports @ isAxiosError.js:12
l.request @ isAxiosError.js:12
r.forEach.l.<computed> @ isAxiosError.js:12
(anonymous) @ isAxiosError.js:12
(anonymous) @ test.html:6
test.html:1 Access to XMLHttpRequest at 'http://localhost:8080/api/v1/keyrequest/?name=app' from origin 'http://localhost:8000' has been blocked by CORS policy: Request header field content-type is not allowed by Access-Control-Allow-Headers in preflight response.
test.html:12 o {message: 'Network Error', name: 'AxiosError', code: 'ERR_NETWORK', config: {…}, request: XMLHttpRequest, …}code: "ERR_NETWORK"config: {transitional: {…}, transformRequest: Array(1), transformResponse: Array(1), timeout: 0, adapter: ƒ, …}message: "Network Error"name: "AxiosError"request: XMLHttpRequest {data: undefined, onreadystatechange: null, readyState: 4, timeout: 0, withCredentials: false, …}response: XMLHttpRequest {data: undefined, onreadystatechange: null, readyState: 4, timeout: 0, withCredentials: false, …}[[Prototype]]: Error
(anonymous) @ test.html:12
Promise.catch (async)
(anonymous) @ test.html:12
VM149:1          POST http://localhost:8080/api/v1/keyrequest/?name=app net::ERR_FAILED
```

And here is the header information for each call:
`keyrequest` preflight:
```
General
Request URL: http://localhost:8080/api/v1/keyrequest/?name=app
Request Method: OPTIONS
Status Code: 200 OK
Remote Address: [::1]:8080
Referrer Policy: strict-origin-when-cross-origin

Response Headers
Access-Control-Allow-Methods: POST
Access-Control-Allow-Origin: *
Content-Language: en
Content-Type: text/html;charset=utf-8
Date: Fri, 30 Sep 2022 03:27:12 GMT
Expires: Thu, 01 Jan 1970 00:00:00 GMT
Server: Jetty(9.4.48.v20220622)
Set-Cookie: JSESSIONID=node01vnpq8kxg6avx1odlteik6lek18.node0; Path=/api
Transfer-Encoding: chunked

Request Headers
Accept: */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9
Access-Control-Request-Headers: content-type
Access-Control-Request-Method: POST
Cache-Control: no-cache
Connection: keep-alive
Host: localhost:8080
Origin: http://localhost:8000
Pragma: no-cache
Referer: http://localhost:8000/
Sec-Fetch-Dest: empty
Sec-Fetch-Mode: cors
Sec-Fetch-Site: same-site
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36
```

`keyrequest` post:
```
General
Request URL: http://localhost:8080/api/v1/keyrequest/?name=app
Referrer Policy: strict-origin-when-cross-origin

Request Headers
Provisional headers are shown
Accept: application/json, text/plain, */*
Content-Type: application/json
Referer: http://localhost:8000/
sec-ch-ua: ".Not/A)Brand";v="99", "Google Chrome";v="103", "Chromium";v="103"
sec-ch-ua-mobile: ?0
sec-ch-ua-platform: "Linux"
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36
``` 

`version` response:
```
General
Request URL: http://localhost:8080/api/v1/version
Request Method: GET
Status Code: 302 
Referrer Policy: strict-origin-when-cross-origin

Response Headers
Content-Language: en
Content-Length: 0
Date: Fri, 30 Sep 2022 03:27:12 GMT
Location: http://localhost:8080/api/v1/version/
Server: Jetty(9.4.48.v20220622)

Request Headers
Accept: application/json, text/plain, */*
Accept-Encoding: gzip, deflate, br
Accept-Language: en-US,en;q=0.9
Cache-Control: no-cache
Connection: keep-alive
Host: localhost:8080
Origin: http://localhost:8000
Pragma: no-cache
Referer: http://localhost:8000/
sec-ch-ua: ".Not/A)Brand";v="99", "Google Chrome";v="103", "Chromium";v="103"
sec-ch-ua-mobile: ?0
sec-ch-ua-platform: "Linux"
Sec-Fetch-Dest: empty
Sec-Fetch-Mode: cors
Sec-Fetch-Site: same-site
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/103.0.0.0 Safari/537.36
```

As expected from the responses going to `http://localhost:8080/manage/` I don't see the API Key in the list.  Navigating to `http://localhost:8080/api/v1/version/` in the browser responds with status 200 and `{"version":"v4.0.7"}`

**Comment by tvbarnette999** [comment]
Thanks for the code-snippet and errors! They saved me a lot of time hunting this down, as the `fetch` calls I was testing with before didn't hit it but the axios calls do in the same browser. There are several issues happening here and down-the-line from these errors once I fixed the original errors:
1. The trailing-slash redirect is not getting the CORS headers - that's the first error you are hitting in your `GET`. (If I add it to the axios call it then gets the same `content-type` error, but the redirect will be fixed) 
2. The `Access-Control-Allow-Headers` isn't getting set
3. `Access-Control-Allow-Headers` can't be * when `Authorization` header is present, so it'll now be set to `Content-Type, If-Modified-Since, Authorization`
4. Axios seems to error-out on a `304 Not Modified` response, which [this issue on their github](https://github.com/axios/axios/issues/703) has a link to more info about handling.

We've got fixes in the works for the redirect CORS headers & the `Access-Control-Allow-Headers`; your code snippet is working as-is in our tests with them. They'll be in the next release (should be in the next 2 weeks or so), but I'm going to re-open the issue and if you get a chance to verify it works for you after the next release you can close it out.

**Comment by jvens** [comment]
I just realized I never responded to this.  This issue has been resolved and is no longer an issue with that last set of changes you made.  Thank you for looking at this, and thank you for providing the API!
